src/main/java/com/ibm/crail/storage/blkdev/BlkDevStorageServer.java

```diff
@@ -38,39 +41,62 @@
 public class BlkDevStorageServer implements StorageServer {
 	private static final Logger LOG = CrailUtils.getLogger();
 
-	private final InetSocketAddress storageAddr;
-	private final Path path;
+	private InetSocketAddress storageAddr;
+	private Path path;
 	private boolean isAlive;
+	private boolean initialized = false;
+	private long addr;
+	private long alignedSize;
+	private BlkDevStorageEndpoint endPoint;
 
 	public BlkDevStorageServer() throws Exception {
+
+	}
+	public void init(CrailConfiguration crailConfiguration, String[] args) throws Exception {
+		if (initialized)	 {
+			throw new IOException("BlkDevStorageTier already initialized");
+		}
+		initialized = true;
+		BlkDevStorageConstants.init(crailConfiguration, args);
 		LOG.info("initalizing block device datanode");
 		// We do not support multiple block devices yet
 		InetAddress address = InetAddress.getLoopbackAddress();
 		int port = 12345;
+
 		storageAddr = new InetSocketAddress(address, port);
 		String directory = BlkDevStorageConstants.DATA_PATH;
 		path = FileSystems.getDefault().getPath(directory);
 		if (!Files.exists(path)) {
 			throw new IllegalArgumentException("BlkDev path does not exists!");
 		}
-		isAlive = false;
+		isAlive = true;
+		alignedSize = BlkDevStorageConstants.STORAGE_LIMIT - (BlkDevStorageConstants.STORAGE_LIMIT % BlkDevStorageConstants.ALLOCATION_SIZE);
+		alignedSize = BlkDevStorageConstants.STORAGE_LIMIT;
+		this.addr = 0;
```

Comment:
I think we can move this to allocateResource() since we do not need it anywhere else